### PR TITLE
br: revert new introduced `BatchScanRegions` to keep compatibility

### LINKED ISSUE #54313: tidb lightning: daily run find  'Lightning is stuck" for hours
[git-hash=b9a31b231a7d9a64da81cb071b3db26fcb55cc38]

toml
```

        [lightning]
        level = "info"
        check-requirements = false
        status-addr = ':8289'
        file = "/tmp/tidb-lightning_1719543717.log"

        [tikv-importer]
        backend = "local"
        incremental-import = false
        sorted-kv-dir = "/tiup/sorted-kv-dir"

        [tidb]
        # Information of the target cluster
        port = 4000
        user = "root"
        password = ""
        host = "tidb-1-peer"
        status-port = 10080
        pd-addr = "pd-peer:2379"

        [mydumper]
        no-schema = true
        data-source-dir = 's3://tmp/test?access-key=minioadmin&secret-access-key=minioadmin&endpoint=http://minio-peer:9000'
        [mydumper.csv]
        header = false

        [checkpoint]
        # Whether to enable checkpoints.
        enable = true
        driver = "file"

        [post-restore]
        checksum = true
```

image

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve?

Issue Number: close #54313 #54341

Problem Summary:

### What changed and how does it work?

We have deprecated the `ScanRegion` and introduce `BatchScanRegions` as a replacement in v8.2. However, to keep compatibility(v8.2 br/lightning should work with previous TiDB clusters), the tools should keep using the old supported `ScanRegion`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

The integration test case which import into 8.1 tidb with this PR's lightning is passed.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/tests/br_key_locked/codec.go [hunk 1]
```diff
@@ -63,7 +63,8 @@ func (c *codecPDClient) ScanRegions(
 		endKey = codec.EncodeBytes(nil, endKey)
 	}
 
-	regions, err := c.Client.BatchScanRegions(ctx, []pd.KeyRange{{StartKey: startKey, EndKey: endKey}}, limit, opts...)
+	//nolint:staticcheck
+	regions, err := c.Client.ScanRegions(ctx, startKey, endKey, limit, opts...)
```
Comment (D3Hunter) [inline br/tests/br_key_locked/codec.go:67]: those can be removed if we have do it in pd `client-go`?

Comment (you06) [reply br/tests/br_key_locked/codec.go:67]: Sorry I don't get your mean, I just revert it to the old version.

**Review by cfzjywxk — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

File: br/tests/br_key_locked/codec.go (hunk 1 above)
Comment (D3Hunter) [reply br/tests/br_key_locked/codec.go:67]: i mean if we have make interface compatible in `client-go` side, we can still use `BatchScanRegions` here

Comment (you06) [reply br/tests/br_key_locked/codec.go:67]: It's a little weird that pd's client handle unimplement error, and the compatibility is made out of pd client. So I switch back to old interface here.

**Pull request merged**